src/server/pfs/server/server_test.go

```diff
@@ -2002,6 +2002,41 @@ func TestGetFile(t *testing.T) {
 	})
 }
 
+func TestGetFileStream(t *testing.T) {
+	c := GetPachClient(t)
+	require.NoError(t, c.CreateRepo("repo"))
+	pfclient, err := c.PfsAPIClient.PutFile(context.Background())
+	require.NoError(t, err)
+	paths := []string{"foo", "bar", "fizz", "buzz"}
+	for _, path := range paths {
+		// Why are we using pfclient.Send instead of one of the PutFileClient interface methods.
+		require.NoError(t, pfclient.Send(&pfs.PutFileRequest{
+			//TODO(kdelga): Aha! this is where the metadata file comes from.
+			File:  pclient.NewFile("repo", "master", path),
+			Value: []byte(path),
+		}))
+	}
+	_, err = pfclient.CloseAndRecv()
+	require.NoError(t, err)
+
+	cis, err := c.ListCommit("repo", "", "", 0)
+	require.Equal(t, 1, len(cis))
+
+	for _, path := range paths {
+		gfr := &pfs.GetFileRequest{
+			File: pclient.NewFile("repo", "master", path),
+			OffsetBytes: 0,
+			SizeBytes: 0,
+		}
+		// TODO(kdelga): change this to only take context and let Recv take the gfr as arg
+		gfsclient, err := c.PfsAPIClient.GetFileStream(context.Background(), gfr)
```

Comment:
Is this kosher to pass the `GetFileRequest` when getting the client and calling `Recv()` without any args. I noticed on `PutFile` it does not require the request in the args, but these method signatures are auto-generated by protobuf, so I just left it to make it work. Lemme know if there's a better way.

Reply:
This is generally kosher, but you should use the context in the client rather than creating a new one (`c.Ctx()`). We tend to write wrappers around these raw grpc methods to make them a little bit nicer to access, so you'd have `c.GetFileStream` that calls `c.PfsAPIClient.GetFileStream`, but using the underlying methods is totally kosher, just a little more unwieldy.

Reply:
I see. I was pattern matching off of TestPutFiles which does not use the context in the client and also uses the raw grpc, but wasn't sure about the Sending/Receiving mechanisms that for PutFiles generates a put-file client with no args and calls Send with a PutFileRequest as an arg but for PutFiles we generate a get-file client with PutFileRequest as an arg but then send no args when we actually call GetFiles.

I was a bit confused by this inconsistency, but chalked it up to the assumption that because the PutFile rpc call takes a stream rather than a message, client generation occurs without any arguments but because GetFiles takes a message, it's client needs to be generated with the message as an arg, and once we get to the point where it's taking a stream rather than a message, the client generation with more closely resemble what PutFile looks like. Just wanted to run this assumption by you.

I've modified TestGetFiles (and all other instances where context.Background is used)to use the context in the PachClient, but will leave the raw grpc calls as is for now.